src/ScaledDomains.Extensions.Caching.MySql/MySqlServerCache.cs

```diff
@@ -80,20 +107,53 @@ public async Task RefreshAsync(string key, CancellationToken token = new Cancell
             ValidateKey(key);
 
             await _databaseOperations.RefreshCacheItemAsync(key, token);
+
+            RunMaintenanceCacheItems();
         }
 
         public void Remove(string key)
         {
             ValidateKey(key);
 
             _databaseOperations.DeleteCacheItem(key);
+
+            RunMaintenanceCacheItems();
         }
 
         public async Task RemoveAsync(string key, CancellationToken token = new CancellationToken())
         {
             ValidateKey(key);
 
             await _databaseOperations.DeleteCacheItemAsync(key, token);
+
+            RunMaintenanceCacheItems();
+        }
+
+        private void RunMaintenanceCacheItems()
+        {
+            if (_lastMaintenanceRun.Add(_maintenanceInterval) < _clock.UtcNow)
+            {
+                Task.Run(async () =>
```

Comment:
on a highly concurrent system many parallel threads could satisfy this condition.

Reply:
i think it wont be a problem. under few lines the logic check again this condition after the current thread acquires the lock.

Reply:
i'm more worried about the high number of tasks created than the concurrency on the database level. Though adding an extra lock around the check could solve it, it might have a performance impact.
a possible solution could be:
- start a one long running task that is well guarded from exceptions and waiting for a ManualResetEventSlim signal
- the default behavior is that the cache operations ensure that the task is running and signal the event
- a future improvement could be that if supported by the host, instead of creating the task by the cache it could be injected as a IHostedService

Reply:
Ok no problem, I'm updating the implementation to use `BackgroundService`.